docs/common-problems.asciidoc

```diff
@@ -0,0 +1,222 @@
+* <<getsockopt-connection-refused,getsockopt: connection refused>>
+* <<target-machine-refused-connection,No connection could be made because the target machine actively refused it>>
+
+
+[float]
+[[ssl-client-fails]]
+==== SSL client fails to connect
+
+The target host might be unreachable or the certificate may not be valid.
+To fix this problem:
+
+. Make sure that the APM Server process on the target host is running and you can connect to it.
+Try to ping the target host to verify that you can reach it from the host running APM Server.
+Then use either `nc` or `telnet` to make sure that the port is available. For example:
++
+[source,shell]
+----
+ping <hostname or IP>
+telnet <hostname or IP> 5044
+----
+
+. Verify that the certificate is valid and that the hostname and IP match.
+. Use OpenSSL to test connectivity to the target server and diagnose problems.
+See the https://www.openssl.org/docs/manmaster/man1/openssl-s_client.html[OpenSSL documentation] for more info.
+
+[float]
+[[cannot-validate-certificate]]
+==== x509: cannot validate certificate for <IP address> because it doesn't contain any IP SANs
+
+This happens because your certificate is only valid for the hostname present in the Subject field.
+To resolve this problem, try one of these solutions:
+
+* Create a DNS entry for the hostname, mapping it to the server's IP.
+* Create an entry in `/etc/hosts` for the hostname. Or, on Windows, add an entry to
+`C:\Windows\System32\drivers\etc\hosts`.
+* Re-create the server certificate and add a SubjectAltName (SAN) for the IP address of the server. This makes the
+server's certificate valid for both the hostname and the IP address.
+
+[float]
+[[getsockopt-no-route-to-host]]
+==== getsockopt: no route to host
+
+This is not an SSL problem. It's a networking problem. Make sure the two hosts can communicate.
+
+[float]
+[[getsockopt-connection-refused]]
+==== getsockopt: connection refused
+
+This is not an SSL problem. Make sure that Logstash is running and that there is no firewall blocking the traffic.
+
+[float]
+[[target-machine-refused-connection]]
+==== No connection could be made because the target machine actively refused it
+
+A firewall is refusing the connection. Check if a firewall is blocking the traffic on the client, the network, or the
+destination host.
+
+[[io-timeout]]
+[float]
+=== I/O Timeout
+
+I/O Timeouts can occur when your timeout settings across the stack are not configured correctly,
+especially when using a load balancer.
+
+You may see an error like the one below in the APM agent logs, and/or a similar error on the APM Server side:
+
+[source,logs]
+----
+[ElasticAPM] APM Server responded with an error:
+"read tcp 123.34.22.313:8200->123.34.22.40:41602: i/o timeout"
+----
+
+To fix this, ensure timeouts are incrementing from the APM agent,
+through your load balancer, to the APM Server.
+
+By default, the agent timeouts are set at 10 seconds, and the server timeout is set at 3600 seconds.
```

Comment:
Did I get this timeout default correct? It's very different from the legacy default of 30s, so I want to make sure I'm not looking at the wrong configuration variable.

Reply:
@elastic/apm-server seems like Brandon is correct, the default value for the read timeout on the APM package is `3600s` not `30s`: https://github.com/elastic/apm-server/blob/6a3bbb6a72ceaa8e743a09a798f797add29d06c8/apmpackage/apm/manifest.yml#L109-L111

Shouldn't that be set to `30s`? This may actually be something that's causing a lot of the requests that are sent to the APM Server to be queued for longer than necessary.

Reply:
Resolving as this is correct per Slack